src/future.cpp

```diff
@@ -245,17 +302,20 @@ namespace detail {
     good since we need a concrete interface at this level. Above code could be simpler.
 */
 
-void async_(std::chrono::milliseconds delay, function<void()> f) {
-    async_([_delay = delay, _f = move(f)]{
-        if (_delay > std::chrono::milliseconds(0))
-        std::this_thread::sleep_for(_delay);
-    _f();
-    });
+task_system& only_task_system() {
+    static task_system instance;
+    return instance;
+}
+
+void async_(std::chrono::system_clock::time_point time_point, function<void()> f) {
+    if (time_point > std::chrono::system_clock::now())
```

Comment:
This check can avoid the clock for immediate calls by checking:
if ((time_point != std::chrono::system_clock::time_point()) && (time_point > std::chrono::system_clock::now()))